### PR TITLE
examples: Separate non/compiling verify runs

### LINKED ISSUE #14874: wasm-cc example uses all available disk space on CI
We need to figure out what to do:

1. Run CI with more disk space
2. Reduce disk requirements by wasm-cc example
3. Free up disk space by deleting artifacts from previous steps. 

For step 3 @phlax has suggested to delete `envoy-docker-images.tar.xz`

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message: examples: Separate non/compiling verify runs
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue] Fix #14874
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/master/api/review_checklist.md):]

### CODE REVIEW

File: .azure-pipelines/pipelines.yml [hunk 1]
```diff
@@ -149,7 +149,7 @@ stages:
               AZP_BRANCH: $(Build.SourceBranch)
 
   - stage: linux_x64
-    dependsOn: ["precheck"]
+    dependsOn: []
```
Comment (phlax) [inline .azure-pipelines/pipelines.yml:152]: revert before dewipping

File: ci/do_ci.sh [hunk 2]
```diff
@@ -451,6 +452,31 @@ elif [[ "$CI_TARGET" == "verify_examples" ]]; then
   umask 027
   chmod -R o-rwx examples/
   ci/verify_examples.sh "*" wasm-cc
+  df -h
+  exit 0
+elif [[ "$CI_TARGET" == "verify_build_examples" ]]; then
+  df -h
+  echo "verify examples..."
```
Comment (phlax) [inline ci/do_ci.sh:459]: separate to a function?

File: ci/do_ci.sh [hunk 3]
```diff
@@ -141,6 +141,31 @@ function run_process_test_result() {
   "${ENVOY_SRCDIR}"/ci/flaky_test/run_process_xml.sh "$CI_TARGET"
 }
 
+function run_ci_verify () {
+  df -h
```
Comment (phlax) [inline ci/do_ci.sh:145]: rbd

File: ci/do_ci.sh [hunk 4]
```diff
@@ -141,6 +141,31 @@ function run_process_test_result() {
...
+      fi
+  done
+  docker images
+  sudo apt-get update -y
+  sudo apt-get install -y -qq --no-install-recommends expect redis-tools
+  export DOCKER_NO_PULL=1
+  umask 027
+  chmod -R o-rwx examples/
+  ci/verify_examples.sh "${@}" || exit
+  df -h
```
Comment (phlax) [inline ci/do_ci.sh:166]: rbd

**Comment by phlax** [comment]
@yanavlasov would you be able to review ?

**Review by yanavlasov — APPROVED** [review]

**Pull request merged**